www/artists/jcowin/js/slideshow.js

```diff
@@ -0,0 +1,28 @@
+var slideIndex = 1;
+showSlides(slideIndex);
+
+// Next/previous controls
+function plusSlides(n) {
+  showSlides(slideIndex += n);
+}
+
+// Thumbnail image controls
+function currentSlide(n) {
+  showSlides(slideIndex = n);
+}
+
+function showSlides(n) {
+  var i;
+  var slides = document.getElementsByClassName("mySlides");
+  var dots = document.getElementsByClassName("dot");
+  if (n > slides.length) {slideIndex = 1}
+  if (n < 1) {slideIndex = slides.length}
+  for (i = 0; i < slides.length; i++) {
+      slides[i].style.display = "none";
+  }
+  for (i = 0; i < dots.length; i++) {
+      dots[i].className = dots[i].className.replace(" active", "");
+  }
+  slides[slideIndex-1].style.display = "block";
+  dots[slideIndex-1].className += " active";
```

Comment:
These lines are throwing Codacy warnings due to the fact that slides and dots aren't defined in this file. While this is addressed in the individual file, the easiest way to avoid the warnings would be to include a default definition of the two arrays which defines them as empty arrays if they don't exist yet, since this will ensure that Codemy knows they aren't objects.